src/utils.js

```diff
@@ -44,6 +46,24 @@ export const formatFileSize = (size) => {
  */
 export const setDataAttribute = (el, name, value) => el.setAttribute(`data-vv-${name}`, value);
 
+/**
+ * Custom parse behavior on top of date-fns parse function.
+ * @param {String} date
+ * @param {String} format
+ * @return {Date|null}
+ */
+export const parseDate = (date, format) => {
+  const parsed = parse(date, format, new Date());
+
+  // if date is not valid or the formatted output after parsing does not match
+  // the string value passed in (avoids overflows)
+  if (!isValid(parsed) || formatDate(parsed, format) !== date) {
+    return null;
+  }
```

Comment:
There is an issue in date-fns, you might want to share your opinion: https://github.com/date-fns/date-fns/issues/549

Reply:
We noticed and discussed and fixed it here: https://github.com/baianat/vee-validate/issues/522